src/core/tests/Test_StatusHandler.py

```diff
@@ -396,6 +396,32 @@ def test_sort_packages_by_classification_and_state(self):
             self.assertEqual(installation_patches_sorted[12]["name"], "test-package-2")  # | Other              | Excluded    |
             self.assertEqual(installation_patches_sorted[13]["name"], "test-package-1")  # | Other              | NotSelected |
 
+    def test_assessment_packages_map(self):
+        patch_count_for_test = 5
+        patch_id = 'python-samba0_2:4.4.5+dfsg-2ubuntu5.4_Ubuntu_16.04'
+        expected_value = {'version': '2:4.4.5+dfsg-2ubuntu5.4', 'classifications': ['Critical'], 'name': 'python-samba0', 'patchId': 'python-samba0_2:4.4.5+dfsg-2ubuntu5.4_Ubuntu_16.04'}
+
+        status_handler = StatusHandler(self.runtime.env_layer, self.runtime.execution_config, self.runtime.composite_logger, self.runtime.telemetry_writer, self.runtime.vm_cloud_type)
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_for_test)
+        status_handler.set_package_assessment_status(test_packages, test_package_versions, 'Critical')
+        self.assertIsNotNone(status_handler._StatusHandler__assessment_packages_map)
```

Comment:
I don't agree with checking a functions internal variables in asserts this way. My preference would be to check the output of the function that's being tested. Which in this instance would be the output of set_package_assessment_status() 

Adding @kjohn-msft to bring this to attention

Reply:
add more assertions for checking if status_file has the new package chagnes

Reply:
We don't need to check StatusHandler's internal variables to verify if packages exist. Line 412 is reading contents from the status file, you can directly check if the packages exist within that file.
Please remove lines 418 to 423 and check for packages within status file using the status file read in line 412.

Please see lines 415, 416, 417 for reference.

Reply:
done, the reference to internal variable is removed